common/flatpak-run.c

```diff
@@ -2227,16 +2227,36 @@ flatpak_run_setup_base_argv (FlatpakBwrap   *bwrap,
                           "--perms", "0700", "--dir", run_dir,
                           "--setenv", "XDG_RUNTIME_DIR", run_dir,
                           "--symlink", "../run", "/var/run",
-                          "--ro-bind", "/sys/block", "/sys/block",
-                          "--ro-bind", "/sys/bus", "/sys/bus",
-                          "--ro-bind", "/sys/class", "/sys/class",
-                          "--ro-bind", "/sys/dev", "/sys/dev",
-                          "--ro-bind", "/sys/devices", "/sys/devices",
                           "--ro-bind-try", "/proc/self/ns/user", "/run/.userns",
                           /* glib uses this like /etc/timezone */
                           "--symlink", "/etc/timezone", "/var/db/zoneinfo",
                           NULL);
 
+  if (access ("/sys/block", R_OK|X_OK) == 0)
```

Comment:
This is really repetitive now. If you're extracting this from the hard-coded `bwrap` parameters, then I'd prefer it to loop over a global array:

```
/* Directories in /sys to share with the sandbox if accessible.
static const char *const sysfs_dirs[] =
{
  "/sys/block",
  ...,
  "/sys/devices"
};

...

  gsize i;

  ...

  for (i = 0; i < G_N_ELEMENTS (sysfs_dirs); i++)
    {
      const char *dir = sysfs_dirs[i];

      if (access (dir, R_OK|X_OK) == 0)
        flatpak_bwrap_add_args (bwrap, "--ro-bind", dir, dir, NULL);
     else
        g_info (...);
    }
```

more like the way `flatpak_run_setup_usr_links()` iterates through `flatpak_abs_usrmerged_dirs`.

Reply:
Thanks for your suggestion, implemented.